### DISCUSSION TITLE
How to export the ZTD estimation?

### DISCUSSION THREAD

**Discussion by millpoint** [groops-devs/groops#285] [General]
hello everyone, how can I get the ZTD (including ZHD and ZWD) estimation from groops?

**Comment by tmayerguerr** [comment]
Hi,
Within [GnssProcessing](https://groops-devs.github.io/groops/html/GnssProcessing.html)->[parametrization](https://groops-devs.github.io/groops/html/gnssParametrizationType.html)->troposphere->outputfileTroposphere you can write out the estimated delays. The output files can be converted with [GnssTroposphere2TropoSinex](https://groops-devs.github.io/groops/html/GnssTroposphere2TropoSinex.html). Hope this answers your question.
regards
Torsten

**Comment by millpoint** [comment]
Hi, Torsten, thanks for your help, it worked.  could you please provide an example of GnssTroposphere2TropoSinex, I'm confused about creating a Troposphere data estimates template.

**Comment by zhedumi** [comment]
Hi,

The creation of TropoSinex files is a bit more complicated if you want to also have standarddeviations with it. But the basic is to just resample them. Basically the writing of the troposinex consists of three steps

1. Resampling of the Tropodata from the GnssProcessing
2. Compute Variances with the Normalequations
3. Compute geoidheights

If you are only interested in the Tropodata you can just ignore the processing step for the covariance estimation.

I have added the script that was used for repro3 to write the troposinex. While the script is rather old it should work as intented or more so I would suggest using it as a template for the processing steps. For it is quiet some time since I converted to troposinex.

[tropoExampleRepro3.xml.zip](https://github.com/user-attachments/files/17188363/tropoExampleRepro3.xml.zip)



Best regards,

Patrick
